src/brogue/IO.c

```diff
@@ -3168,19 +3339,44 @@ void messageWithoutCaps(char *msg, enum messageFlags flags) {
         }
     }*/
 
-    for (i = MESSAGE_LINES - 1; i >= 1; i--) {
-        messageConfirmed[i] = messageConfirmed[i-1];
-        strcpy(displayedMessage[i], displayedMessage[i-1]);
+    // Add the message to the archive, bumping counts for recent duplicates
+    newMessage = true;
+
+    // For each at most MESSAGE_ARCHIVE_ENTRIES - 1 past entries..
+    for (i = 1; i < MESSAGE_ARCHIVE_ENTRIES; i++) {
+        archiveEntry = getArchivedMessage(i);
+
+        // Consider messages that arrived this turn for collapsing.  Also
+        // consider the latest entry (which may be from past turns) if the
+        // incoming message is not semi-colon foldable.
+        if ((i > 1 || flags & FOLDABLE) && archiveEntry->turn != rogue.playerTurnNumber) {
```

Comment:
The comment and the if are written inverted to each other, it's not easy to parse

Reply:
I will invert the conditional with an explicit negation.